src/mscorlib/corefx/System/Globalization/TextInfo.Unix.cs

```diff
@@ -17,46 +20,76 @@ public partial class TextInfo
         internal unsafe TextInfo(CultureData cultureData)
         {
             // TODO: Implement this fully.
-            this.m_cultureData = cultureData;
-            this.m_cultureName = this.m_cultureData.CultureName;
-            this.m_textInfoName = this.m_cultureData.STEXTINFO;
+            m_cultureData = cultureData;
+            m_cultureName = m_cultureData.CultureName;
+            m_textInfoName = m_cultureData.STEXTINFO;
+            m_needsTurkishCasing = NeedsTurkishCasing(this.m_textInfoName);
         }
 
+        [System.Security.SecuritySafeCritical]
         private unsafe string ChangeCase(string s, bool toUpper)
         {
-            Contract.Assert(s != null);              
-            // TODO: Implement this fully.
+            Contract.Assert(s != null);
 
-            StringBuilder sb = new StringBuilder(s.Length);
+            char[] dstBuf = new char[s.Length];
 
-            for (int i = 0; i < s.Length; i++)
+            fixed (char* pSrc = s, pDst = dstBuf)
```

Comment:
I will use the `StringBuilderCache`.  I will wish there was a way to call the string's .ctor and provide a filler function to initialize the memory, so we could save the allocation and copy.  Hopefully someday the framework can provide that.